### PR TITLE
VZ-8194 VZ-8196 service template

### PR DESCRIPTION
**Pull request by mbenmclean**
Documents new support for service template to customize Helidon workload service

### CODE REVIEW

File: content/en/docs/applications/workloads/helidon/helidon.md [hunk 1]
```diff
@@ -83,6 +83,48 @@ developing and deploying the Verrazzano Helidon application.
...
 For more Verrazzano Helidon application examples, see [Examples]({{< relref "/docs/samples/_index.md" >}}).
 
+### Customizing Helidon Workload service
+
+By default, deploying a Helidon Workload will create a service to provide ingress into the Helidon pod. That service 
+can be customized by added a `ServiceTemplate` to the `VerrazzanoHelidonWorkload` spec. This supports customizing the
+service `metadata` and `serviceSpec` sections to do things like add labels and annotations, customize the service ports,
+modify the pod selector, etc. It is not necessary to provide complete `metadata` and `serviceSpec` sections. Just add the fields you
+would like to customize and Verrazzano will create the rest of the fields based the information
+provided in the `DeploymentTemplate`.
```
Comment (pfmackin) [inline content/en/docs/applications/workloads/helidon/helidon.md:93]: Do you want to say anything about the protocol stuff you just fixed?

Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md:93]: What do you mean? Like the servicePort name prefix?

File: content/en/docs/applications/workloads/helidon/helidon.md [hunk 2]
```diff
@@ -83,6 +83,48 @@ developing and deploying the Verrazzano Helidon application.
 
 For more Verrazzano Helidon application examples, see [Examples]({{< relref "/docs/samples/_index.md" >}}).
 
+### Customizing Helidon Workload service
+
+By default, deploying a Helidon Workload will create a service to provide ingress into the Helidon pod. That service
```
Comment (pfmackin) [inline content/en/docs/applications/workloads/helidon/helidon.md:88]: Service doesn't provide ingress, the gateway does.   The service name is used by the VirtualService host field to specify the destination of the request.

Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md:88]: Ok will reword

**Review by pfmackin — APPROVED** [review]

File: content/en/docs/applications/workloads/helidon/helidon.md [hunk 3]
```diff
@@ -83,6 +83,48 @@ developing and deploying the Verrazzano Helidon application.
 
 For more Verrazzano Helidon application examples, see [Examples]({{< relref "/docs/samples/_index.md" >}}).
 
+### Customizing Helidon Workload service
+
+By default, deploying a Helidon Workload will create a service to access the Helidon pod. That service 
+can be customized by added a `ServiceTemplate` to the `VerrazzanoHelidonWorkload` spec. This supports customizing the
```
Comment (kminder) [inline content/en/docs/applications/workloads/helidon/helidon.md:89]: Should this be serviceTemplate (i.e. lower case s)?

**Review by kminder — APPROVED** [review]
Approved.  My comment about serviceTemplate can be addressed along with any feedback from the doc team.

File: content/en/docs/applications/workloads/helidon/helidon.md (hunk 1 above)
Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md:93]: I don't think we should cover this in our documentation.  We could link to the Istio documentation I suppose. What do you think?

File: content/en/docs/applications/workloads/helidon/helidon.md (hunk 3 above)
Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md:89]: I'm not sure. @rosemarymarano what do you think?

Comment (rosemarymarano) [reply content/en/docs/applications/workloads/helidon/helidon.md]: Yes, it should be written as it is in the code text, `serviceTemplate`.

File: content/en/docs/applications/workloads/helidon/helidon.md (hunk 1 above)
Comment (rosemarymarano) [reply content/en/docs/applications/workloads/helidon/helidon.md]: If you think that certain other documentation is needed, but it is covered in the Istio docs, then the correct thing to do would be to add a link to it.

**Review by rosemarymarano — APPROVED** [review]
I added my edits (minor) and answered the questions posed to me. I'm approving this PR. If you decide that it needs further modification, please let me know.

File: content/en/docs/applications/workloads/helidon/helidon.md (hunk 1 above)
Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md]: I added it

File: content/en/docs/applications/workloads/helidon/helidon.md (hunk 3 above)
Comment (mbenmclean) [reply content/en/docs/applications/workloads/helidon/helidon.md]: fixed

File: content/en/docs/applications/workloads/helidon/helidon.md [hunk 4]
```diff
@@ -93,7 +93,9 @@ can be customized by added a `serviceTemplate` to the `VerrazzanoHelidonWorkload
 service `metadata` and `serviceSpec` sections to do things, like add labels and annotations, customize the service ports,
 modify the pod selector, and such. It is not necessary to provide complete `metadata` and `serviceSpec` sections. Just add the fields you
 would like to customize and Verrazzano will create the rest of the fields, based the information
-provided in the `deploymentTemplate`.
+provided in the `deploymentTemplate`. Please refer to the Istio [Protocol Selection](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection)
+documentation when customizing service ports.
```
Comment (rosemarymarano) [inline content/en/docs/applications/workloads/helidon/helidon.md:97]: -> When customizing service ports, refer to [Protocol Selection](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection) in the Istio documentation.

**Pull request merged**